ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -233,6 +281,25 @@ let all (lookup: string -> string option) (list: string -> string list) ~__conte
 	    Db.VM_guest_metrics.set_other ~__context ~self:gm ~value:other;
 	    Helpers.call_api_functions ~__context (fun rpc session_id -> Client.Client.VM.update_allowed_operations rpc session_id self);
 	  end;
+	  if(vif_state_cached <> vif_state) then begin
```

Comment:
Comparing vif_state with vif_state_cached is more costly than doing a logical AND across one of them.

But as I said, I see no need to build up vif_state in the first place: we can do logical AND as we go along and cache the result.